### PR TITLE
Fix help page images

### PR DESCRIPTION
**Pull request by cwarnermm**
#### Summary
Updated in-product help documentation to fix broken links and to correct outdated information.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-36399

#### Release Note
```release-note
Updated in-product help documentation to fix broken links and to correct outdated information.
```

### CODE REVIEW

File: components/help/components/attaching.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export default function Attaching(): JSX.Element {
             <p>
                 <FormattedMessage
                     id='help.attaching.methods.description'
-                    defaultMessage='Attach a file by drag and drop or by clicking the attachment icon in the message input box.'
+                    defaultMessage='You have three ways to attach a file: by dragging and dropping, using the Attachment icon, or by copying and pasting.'
```
Comment (cwarnermm) [inline components/help/components/attaching.tsx:29]:
```suggestion
                    defaultMessage='You have three ways to attach a file: dragging and dropping, using the Attachment icon, or by copying and pasting.'
```

**Comment by cwarnermm** [comment]
@justinegeffen - It would be great to review these updates contextually in a build. Is there a way to spin up a test build of this PR for review purposes?

File: components/help/components/commands.tsx [hunk 2]
```diff
@@ -22,57 +22,26 @@ export default function HelpCommands(): JSX.Element {
...
+                    defaultMessage='You can execute commands by typing into the text input box to perform operations in Mattermost. These commands are called slash commands. To run a slash command, type `/` followed by a command and some arguments to perform actions.'
                 />
             </p>
 
             <p>
                 <FormattedMarkdownMessage
                     id='help.commands.intro2'
-                    defaultMessage='Built-in slash commands come with all Mattermost installations and custom slash commands are configurable to interact with external applications. Learn about configuring custom slash commands on the [developer slash command documentation page](!http://docs.mattermost.com/developer/slash-commands.html).'
+                    defaultMessage='Built-in slash commands come with all Mattermost installations. See the [product documentation](!https://docs.mattermost.com/messaging/executing-slash-commands.html#built-in-commands) for a list of available built-in slash commands. Custom slash commands can interact with external applications.'
                 />
```
Comment (calebroseland) [inline components/help/components/commands.tsx:33]:
```suggestion
                />
            </p>
```

**Review by calebroseland — COMMENTED** [review]
I think these are why the build is failing

File: components/help/components/formatting.tsx [hunk 3]
```diff
@@ -258,15 +258,8 @@ export default function HelpFormatting(): JSX.Element {
...
+                    defaultMessage='Create in-line images using an `!` followed by the alt text in square brackets and the link in normal brackets. Add hover text by placing it in quotes after the link. See the [product documentation](!https://docs.mattermost.com/messaging/formatting-text.html#in-line-images) for details on working with in-line images.'
                 />
-            </p>
-            <FormattedMessage
-                id='help.formatting.imagesExample'
-                defaultMessage={'![alt text]({siteUrl}/static/images/logo-email.png "Mattermost horizontal logo")\n\nand\n\n[![Atlassian](https://wac-cdn.atlassian.com/assets/img/favicons/atlassian/favicon.png)](https://travis-ci.org/mattermost/mattermost-server) [![Github](https://github.githubassets.com/favicon.ico)](https://github.com/mattermost/mattermost-server)'}
-                values={{siteURL: getSiteURL()}}
-            >
-                {(example) => renderRawExampleWithResult(example)}
             </FormattedMessage>
```
[the remark this answers is not in the document]
Comment (hahmadia) [reply components/help/components/formatting.tsx]: In addition to these, make sure you re generate the i18n files.

I can't call off top of my head but you need to run 'make i18n-check' . That's also why the build is failing.

**Comment by hahmadia** [comment]
Hey Carrie,

I would suggest you run the following two commands and push the changes

1. make fix-style
2. make i18n-extract

then push changes.

**Comment by cwarnermm** [comment]
@hahmadia Thank you so much for helping with this ticket! I don't have a local development environment set up for the webapp repository. Given this, is there a way for me to perform the steps you list above? Or would these steps need to be performed by someone with a dev environment?

**Comment by cwarnermm** [comment]
After running the 2 make targets that @hahmadia recommended, 1 linter error was returned:
```
.../mattermost-webapp/components/help/components/formatting.tsx
  10:9  error  'getSiteURL' is defined but never used  @typescript-eslint/no-unused-vars
```

This error doesn't appear to be related to my changes, so I've left this line of the source file as-is.

**Comment by calebroseland** [comment]
@cwarnermm seems like `getSiteURL` was used in a code block that you removed, so it's just a linter-nag to remove the import up at the top since it's not used anywhere else,

**Comment by cwarnermm** [comment]
Thanks, @calebroseland! I was wrong - I _did_ touch things to result in that linter error :) This is a very good learning experience walking through this process end to end. So glad to have your support along the way!

**Comment by calebroseland** [comment]
The linter's usually right 😅. For everything else, there's `// eslint-disable-line` 😛

**Review by hahmadia — APPROVED** [review]

**Comment by calebroseland** [comment]
@cwarnermm It's _really_ subtle, but this last blocker `Should not have do-not-merge/release-note-label-needed label` is due to the release note codeblock needing to be tagged in the description

CleanShot 2021-07-23 at 11 59 54@2x

**Review by calebroseland — APPROVED** [review]

**Comment by cwarnermm** [comment]
Thanks to ongoing support from @calebroseland and @hahmadia, the in-product help text has been updated to:
- fix broken image links
- update the text for clarity and to match current product functionality

@justinegeffen @esethna - A preview test server is available to help with review context.

File: components/help/components/composing.tsx [hunk 4]
```diff
@@ -50,61 +50,61 @@ export default function HelpComposing(): JSX.Element {
             <p>
                 <FormattedMessage
                     id='help.composing.replies.description1'
-                    defaultMessage='Reply to a message by clicking the reply icon next to any message text. This action opens the right-hand sidebar where you can see the message thread, then compose and send your reply. Replies are indented slightly in the center pane to indicate that they are child messages of a parent post.'
+                    defaultMessage='Select the **Reply** icon next to any message to open the right-hand sidebar to respond to a thread. When Collapsed Reply Threads (Beta) is disabled, replies are indented slightly in the center pane to indicate that they are child messages of a parent post. When Collapsed Reply Threads (Beta) is enabled, replies only display in the right-hand sidebar.'
```
Comment (esethna) [inline components/help/components/composing.tsx:53]: Propose we don't include anything about CRT in here yet

**Review by esethna — APPROVED** [review]

**Comment by esethna** [comment]
Thanks Carrie! Will leave it in docs team hands to finish the reviews

**Comment by cwarnermm** [comment]
@justinegeffen There are a couple of instances where bolded text on the Composing page is displaying as markdown rather than as bold text. A fresh set of eyes would be most welcome on this!

File: components/help/components/attaching.tsx [hunk 5]
```diff
@@ -26,87 +26,87 @@ export default function Attaching(): JSX.Element {
             <p>
                 <FormattedMessage
                     id='help.attaching.methods.description'
-                    defaultMessage='Attach a file by drag and drop or by clicking the attachment icon in the message input box.'
+                    defaultMessage='You have three ways to attach a file: dragging and dropping files, using the Attachment icon, or by copying and pasting files.'
```
Comment (justinegeffen) [inline components/help/components/attaching.tsx:29]:
```suggestion
                    defaultMessage='There are three ways to attach a file: You can drag and drop files, use the Attachment icon, or copy and paste files.'
```

File: i18n/en.json [hunk 6]
```diff
@@ -3184,52 +3184,51 @@
...
+  "help.attaching.dragdrop.title": "Drag and Drop Files",
+  "help.attaching.icon.description": "Alternatively, upload files by selecting the **Attachment** icon inside the message input box. In the system file viewer, navigate to the desired files, then select **Open** to upload them to the message input box. Optionally type a message, then press **ENTER** to post the message.",
+  "help.attaching.icon.title": "Use the Attachment Icon",
+  "help.attaching.limitations.description": "Mattermost supports up to five files attached per post. The default maximum file size is 100 MB (megabytes), but this can be changed by your System Admin. Image files can be a maximum size of 6048 pixels x 4032 pixels, or 24 MP (mega pixels), or a raw image file size of approximately 36 MB.",
   "help.attaching.limitations.title": "File Size Limitations",
-  "help.attaching.methods.description": "Attach a file by drag and drop or by clicking the attachment icon in the message input box.",
+  "help.attaching.methods.description": "You have three ways to attach a file: dragging and dropping files, using the Attachment icon, or by copying and pasting files.",
```
Comment (justinegeffen) [inline i18n/en.json:3195]:
```suggestion
  "help.attaching.methods.description": "There are three ways to attach a file: You can drag and drop files, use the Attachment icon, or copy and paste files.",
```

**Review by justinegeffen — APPROVED** [review]
One suggestion but otherwise this is an awesome PR, really great work!

**Comment by justinegeffen** [comment]
> @justinegeffen There are a couple of instances where bolded text on the Composing page is displaying as markdown rather than as bold text. A fresh set of eyes would be most welcome on this!

I didn't see anything that stood out at me but I may have missed something. Do you have an example and is this in the file itself or the preview?

**Comment by cwarnermm** [comment]
image

**Pull request merged**